src/components/Filter/Filter.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { useProduct } from '../../context/product-context'
+import "./Filter.css"
+function Filter() {
+    
+    const {state:{sortByPrice , sortByRating , sortByCategory:{mobile,laptop} , priceRange} ,dispatch} = useProduct()
+    
+  return (
+    <nav className="sidebar-nav">
+            <div className="sidebar-container flex space-between align-center">
+                <h3 className="sidebar-heading">Filters</h3>
+                <span className="sidebar-clear-text clear-all" onClick={() => dispatch({type: "CLEAR"})}>Clear</span>
+            </div>
+            <div className="sidebar-container">
+                <h3 className="sidebar-heading">Price</h3>
+                <p className="flex space-between align-center"><span className="range-min">5000</span><span className="range-max">{priceRange}</span></p>
+                <input type="range" className="input" min={5000} max={50000} value = {priceRange} step={5000} onChange={(event) => dispatch({type: "PRICE", payload: event.target.value})}/>
+                
+            </div>
+            <div className="sidebar-container">
+                <h3 className="sidebar-heading">Categories</h3>
+                <div className="list-container">    
+                    <ul className="list">
+                        <li className="list-item"><label htmlFor="mobile"><input type="checkbox" id="mobile" className="check-input" checked={mobile} onChange={() => dispatch({type: "MOBILE" })}/>Mobile</label></li>
+                        <li className="list-item"><label htmlFor="laptop"><input type="checkbox" id="laptop" className="check-input" checked={laptop}
+                        onChange={() => dispatch({type: "LAPTOP"})}/>Laptop</label></li>
```

Comment:
you can use word-wrap before pushing code for good readability.

Reply:
ok